apps/game_backend/lib/game_backend/matches.ex

```diff
@@ -45,34 +46,20 @@ defmodule GameBackend.Matches do
     end)
   end
 
-  defp give_trophies(multi, results) do
-    currency_config = Application.get_env(:game_backend, :currencies_config)
+  defp give_prestige(multi, results) do
+    prestige_config = Application.get_env(:game_backend, :arena_prestige)
 
-    Enum.reduce(results, multi, fn result, multi ->
-      Multi.run(
-        multi,
-        {:add_trophies_to, result["user_id"]},
-        fn _, %{get_google_users: google_users} ->
-          google_user = Enum.find(google_users, fn google_user -> google_user.id == result["user_id"] end)
-
-          amount_of_trophies =
-            Enum.find(google_user.user.currencies, fn user_currency -> user_currency.currency.name == "Trophies" end)
-            |> case do
-              nil -> 0
-              currency -> currency.amount
-            end
-
-          amount =
-            get_amount_of_trophies_to_modify(amount_of_trophies, result["position"], currency_config)
-
-          Currencies.add_currency_by_name_and_game!(
-            google_user.user.id,
-            "Trophies",
-            Utils.get_game_id(:curse_of_mirra),
-            amount
-          )
-        end
-      )
+    Enum.reduce(results, multi, fn result, transaction_acc ->
+      Multi.run(transaction_acc, {:update_prestige, result["user_id"]}, fn repo, %{get_google_users: google_users} ->
+        google_user = Enum.find(google_users, fn google_user -> google_user.id == result["user_id"] end)
```

Comment:
What if the user is a guest (didn't link google account yet)?
I think we should start working on User and not on GoogleUser so it works fine for both.

Reply:
> What if the user is a guest (didn't link google account yet)?

Is that an actual thing that will happen? Also, we need to fix the match results processing to handle this "bad"/missing users. But I will make a follow up ticket for it, that will touch to many things here and don't want to overcomplicate the PR

> I think we should start working on User and not on GoogleUser so it works fine for both.

Yep, this is also a mistake from the match result inception. I'll include it in the ticket to fix all this